docs/source/developers/guide/documentation.rst

```diff
@@ -27,4 +27,79 @@
 
 **************************
 Helping with documentation
-**************************
\ No newline at end of file
+**************************
+
+**A great way to contribute to the project is to improve
+documentation.**
+
+If you are an Arrow user and you found some docs to be
+incomplete or inaccurate, share your hard-earned knowledge
+with the rest of the community.
+
+If you didn't come across something to improve in the
+documentation itself, you can search for an issue on JIRA.
+
+.. note::
+   When searching for JIRA issue that deals with documentation
+   try selecting **Components** from **More** tab in JIRA search
+   and there find **Documentation** from the list.
+
+   See `Example search. <https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC>`_
+
+   .. figure:: jira_search_component.jpeg
+      :scale: 40 %
+      :alt: selecting Components in JIRA search
+
+      First select Components tab in JIRA.
+
+   .. figure:: jira_search_documentation.jpeg
+      :scale: 40 %
+      :alt: selecting Documentation in JIRA search
+
+      Then choose Documentation from the Components list.
+
+Documentation improvements are also a great way to gain some
+experience with our submission and review process without
+requiring a lot of local development environment setup. 
+
+.. note::
+   Many documentation-only changes can be made directly in the
+   GitHub web interface by clicking the **Edit this page**
+   on the right top corner of the documentations page. This
+   will handle making a fork and a pull request for you.
+
+   .. figure:: edit_page.jpeg
+      :scale: 20 %
+      :alt: click on edit this page
+
+      On the right corner of the file in GitHub click on pen icon.
+
+   .. figure:: github_edit_page.jpeg
+      :scale: 30 %
+      :alt: edit file in GitHub.
+
+      Now you can edit the file in GitHub.
+
+You could also build the entire project, make the change locally on
+your branch and make the PR this way. But it is by no means superior
+to simply editing via GitHub.
+
+If you wish to build the documentation also, follow detailed instructions
+on :ref:`building-docs`.
+
+Where to find the correct file to change?
+-----------------------------------------
+
+Most of the documentation is located in the ``docs/source`` of the Arrow
+library. Source folder includes:
```

Comment:
Now you added this list, I think it is relevant to mention "docstrings" for Python as well -> for editing the reference docs, you will need to edit the docstring the pyarrow package itself

Reply:
And actually for C++ it's the same issue

Reply:
Ok, will do.